src/modules/75-cf/components/PipelineSteps/FlagConfigurationStep/__tests__/FlagConfigurationStepWidget.test.tsx

```diff
@@ -57,41 +64,154 @@ const renderComponent = (props: Partial<FlagConfigurationStepWidgetProps> = {}):
+      renderComponent()
+
+      expect(spyGetAllFeatures).toBeCalled()
+      expect(spyGetFeatureFlag).toBeCalled()
+      expect(document.getElementsByName('name')[0]).toHaveValue('Step 1 Test')
+      expect(document.getElementsByName('spec.environment')[0]).toHaveValue('Mock Environment')
+      expect(document.getElementsByName('spec.feature')[0]).toHaveValue('Test Flag 1')
+    })
+
+    test('It should prepend the saved flag on to the Select Flag MultitypeInput options when saved flag is not in first page of results', async () => {
+      const savedFlagId = 'Test_Paging_Flag'
+      const pagedResponse = {
+        ...mockFeatureFlags,
+        features: mockFeatureFlags.features.slice(0, CF_DEFAULT_PAGE_SIZE)
+      }
+      const mockFeatureFlag = mockFeatureFlags.features.find(flag => flag.identifier === savedFlagId)
+
+      spyGetAllFeatures.mockReturnValue({ data: pagedResponse, loading: false, error: null, refetch: jest.fn() } as any)
+      spyGetFeatureFlag.mockReturnValue({
+        data: mockFeatureFlag,
+        loading: false,
+        error: null,
+        refetch: jest.fn()
+      } as any)
+
+      renderComponent({
+        initialValues: {
+          type: 'FlagConfiguration',
+          name: 'Step 1 Test',
+          identifier: 'Step_1_Test',
+          spec: {
+            feature: savedFlagId,
+            environment: 'Mock_Environment'
+          },
+          timeout: '10m'
+        }
+      })
+      const flagInput = document.getElementsByName('spec.feature')[0]
+
+      expect(document.getElementsByName('name')[0]).toHaveValue('Step 1 Test')
+      expect(document.getElementsByName('spec.environment')[0]).toHaveValue('Mock Environment')
+      expect(flagInput).toHaveValue('Test Paging Flag')
+
+      // Click the flag dropdown to display options
+      userEvent.click(flagInput)
+      const dropdownOptions = document.getElementsByTagName('li')
+
+      // saved flag should be prepended to list
+      expect(dropdownOptions).toHaveLength(CF_DEFAULT_PAGE_SIZE + 1)
+      expect(dropdownOptions[0]).toHaveTextContent('Test Paging Flag')
+    })
+
+    test('It should NOT prepend the saved flag when the flag is in the first page of results', async () => {
+      // mock array with length of 3
+      const mockFeatures = {
+        ...mockFeatureFlags,
+        features: mockFeatureFlags.features.slice(-3)
+      }
+
+      spyGetAllFeatures.mockReturnValue({ data: mockFeatures, loading: false, error: null, refetch: jest.fn() } as any)
+
+      renderComponent({
+        initialValues: {
+          type: 'FlagConfiguration',
+          name: 'Step 1 Test',
+          identifier: 'Step_1_Test',
+          spec: {
+            feature: 'X_Flag_11', // second flag in mockFeatures array
+            environment: 'Mock_Environment'
+          },
+          timeout: '10m'
+        }
+      })
+
+      const flagInput = document.getElementsByName('spec.feature')[0]
```

Comment:
`getElementsByName` is such a pain when you just want a single thing. I prefer good old `querySelector`
```suggestion
      const flagInput = document.querySelector('[name="spec.feature"]')
```

Reply:
I prefer that too, for some reason I thought it was too 'old school' to use